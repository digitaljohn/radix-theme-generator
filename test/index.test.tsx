/**
 * @jest-environment jsdom
 */

import {
  generateRadixColors,
  getBackgroundColorCss,
  getColorScaleCss,
} from "../src";

const accentValue = "#3D63DD";
const grayValue = "#8B8D98";
const lightBgValue = "#FFFFFF";
const darkBgValue = "#111111";

describe("Basic functionality", () => {
  it("should generate a light theme", async () => {
    const result = generateRadixColors({
      appearance: "light",
      accent: accentValue,
      gray: grayValue,
      background: lightBgValue,
    });

    const css = getColorScaleCss({
      isDarkMode: false,
      name: "blue",
      contrast: result.accentContrast,
      scale: result.accentScale,
      scaleWideGamut: result.accentScaleWideGamut,
      scaleAlpha: result.accentScaleAlpha,
      scaleAlphaWideGamut: result.accentScaleAlphaWideGamut,
      surface: result.accentSurface,
      surfaceWideGamut: result.accentSurfaceWideGamut,
    });

    expect(css).toMatchSnapshot();
  });

  it("should generate a dark theme", async () => {
    const result = generateRadixColors({
      appearance: "dark",
      accent: accentValue,
      gray: grayValue,
      background: darkBgValue,
    });

    const css = getColorScaleCss({
      isDarkMode: true,
      name: "blue",
      contrast: result.accentContrast,
      scale: result.accentScale,
      scaleWideGamut: result.accentScaleWideGamut,
      scaleAlpha: result.accentScaleAlpha,
      scaleAlphaWideGamut: result.accentScaleAlphaWideGamut,
      surface: result.accentSurface,
      surfaceWideGamut: result.accentSurfaceWideGamut,
    });

    expect(css).toMatchSnapshot();
  });

  it("should generate grey scales", async () => {
    const result = generateRadixColors({
      appearance: "light",
      accent: accentValue,
      gray: grayValue,
      background: darkBgValue,
    });

    const css = getColorScaleCss({
      isDarkMode: false,
      name: "gray",
      contrast: "#fff",
      scale: result.grayScale,
      scaleWideGamut: result.grayScaleWideGamut,
      scaleAlpha: result.grayScaleAlpha,
      scaleAlphaWideGamut: result.grayScaleAlphaWideGamut,
      surface: result.graySurface,
      surfaceWideGamut: result.graySurfaceWideGamut,
    });

    expect(css).toMatchSnapshot();
  });

  it("should generate background value", async () => {
    const result = generateRadixColors({
      appearance: "light",
      accent: accentValue,
      gray: grayValue,
      background: darkBgValue,
    });

    const css = getBackgroundColorCss({
      isDarkMode: false,
      background: result.background,
    });

    expect(css).toMatchSnapshot();
  });
});
