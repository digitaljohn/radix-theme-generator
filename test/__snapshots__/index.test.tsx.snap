// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic functionality should generate a dark theme 1`] = `
".dark, .dark-theme {
  --blue-1: #0c111c;
  --blue-2: #111725;
  --blue-3: #172448;
  --blue-4: #1d2e61;
  --blue-5: #243974;
  --blue-6: #2d4484;
  --blue-7: #375098;
  --blue-8: #405eb2;
  --blue-9: #3d63dd;
  --blue-10: #3f5cb0;
  --blue-11: #93b4ff;
  --blue-12: #d5e2ff;

  --blue-a1: #0012fb0c;
  --blue-a2: #1156f916;
  --blue-a3: #2b64ff3b;
  --blue-a4: #3567ff56;
  --blue-a5: #3f71fd6b;
  --blue-a6: #4b7afd7c;
  --blue-a7: #5480ff91;
  --blue-a8: #5783ffad;
  --blue-a9: #4571ffdb;
  --blue-a10: #5580feab;
  --blue-a11: #93b4ff;
  --blue-a12: #d5e2ff;

  --blue-contrast: #fff;
  --blue-surface: #111d3980;
  --blue-indicator: #3d63dd;
  --blue-track: #3d63dd;
}

@supports (color: color(display-p3 1 1 1)) {
  @media (color-gamut: p3) {
    .dark, .dark-theme {
      --blue-1: oklch(17.8% 0.0247 266.8);
      --blue-2: oklch(20.7% 0.0302 266.8);
      --blue-3: oklch(27.1% 0.0694 266.8);
      --blue-4: oklch(31.8% 0.0933 266.8);
      --blue-5: oklch(36.1% 0.1046 266.8);
      --blue-6: oklch(40.4% 0.1106 266.8);
      --blue-7: oklch(45% 0.1203 266.8);
      --blue-8: oklch(50.3% 0.1373 266.8);
      --blue-9: oklch(54.3% 0.1913 266.8);
      --blue-10: oklch(49.7% 0.1373 266.8);
      --blue-11: oklch(77.7% 0.1234 266.8);
      --blue-12: oklch(91.1% 0.0428 266.8);

      --blue-a1: color(display-p3 0 0.0706 0.9804 / 0.043);
      --blue-a2: color(display-p3 0.1176 0.3608 1 / 0.08);
      --blue-a3: color(display-p3 0.2275 0.4039 1 / 0.223);
      --blue-a4: color(display-p3 0.2627 0.4196 1 / 0.324);
      --blue-a5: color(display-p3 0.302 0.451 1 / 0.4);
      --blue-a6: color(display-p3 0.3451 0.4902 1 / 0.467);
      --blue-a7: color(display-p3 0.3725 0.5098 1 / 0.547);
      --blue-a8: color(display-p3 0.3843 0.5255 1 / 0.652);
      --blue-a9: color(display-p3 0.3176 0.451 1 / 0.824);
      --blue-a10: color(display-p3 0.3843 0.5176 1 / 0.643);
      --blue-a11: color(display-p3 0.6196 0.7216 1 / 0.975);
      --blue-a12: color(display-p3 0.8549 0.898 1 / 0.988);

      --blue-contrast: #fff;
      --blue-surface: color(display-p3 0.0706 0.1098 0.2118 / 0.5);
      --blue-indicator: oklch(54.3% 0.1913 266.8);
      --blue-track: oklch(54.3% 0.1913 266.8);
    }
  }
}"
`;

exports[`Basic functionality should generate a light theme 1`] = `
":root, .light, .light-theme {
  --blue-1: #fdfdfe;
  --blue-2: #f7f9ff;
  --blue-3: #edf2fe;
  --blue-4: #dfeaff;
  --blue-5: #d0dfff;
  --blue-6: #bdd1ff;
  --blue-7: #a6bff9;
  --blue-8: #87a5ef;
  --blue-9: #3d63dd;
  --blue-10: #3657c3;
  --blue-11: #395bc7;
  --blue-12: #1d2e5c;

  --blue-a1: #00008002;
  --blue-a2: #0040ff08;
  --blue-a3: #0047f112;
  --blue-a4: #0058ff20;
  --blue-a5: #0052ff2f;
  --blue-a6: #004eff42;
  --blue-a7: #0048ee59;
  --blue-a8: #0040dd78;
  --blue-a9: #0032d2c2;
  --blue-a10: #002ab3c9;
  --blue-a11: #002cb7c6;
  --blue-a12: #001347e2;

  --blue-contrast: #fff;
  --blue-surface: #f5f8ffcc;
  --blue-indicator: #3d63dd;
  --blue-track: #3d63dd;
}

@supports (color: color(display-p3 1 1 1)) {
  @media (color-gamut: p3) {
    :root, .light, .light-theme {
      --blue-1: oklch(99.4% 0.0015 266.8);
      --blue-2: oklch(98.3% 0.008 266.8);
      --blue-3: oklch(96.1% 0.0172 266.8);
      --blue-4: oklch(93.5% 0.0338 266.8);
      --blue-5: oklch(90.3% 0.0512 266.8);
      --blue-6: oklch(86.3% 0.0714 266.8);
      --blue-7: oklch(80.7% 0.0869 266.8);
      --blue-8: oklch(73% 0.1134 266.8);
      --blue-9: oklch(54.3% 0.1913 266.8);
      --blue-10: oklch(49.7% 0.1729 266.8);
      --blue-11: oklch(51% 0.1729 266.8);
      --blue-12: oklch(31.3% 0.0856 266.8);

      --blue-a1: color(display-p3 0.0196 0.0196 0.5098 / 0.008);
      --blue-a2: color(display-p3 0.0196 0.2667 0.8784 / 0.032);
      --blue-a3: color(display-p3 0.0078 0.2392 0.8863 / 0.067);
      --blue-a4: color(display-p3 0.0039 0.3059 0.9373 / 0.118);
      --blue-a5: color(display-p3 0.0039 0.2784 0.9333 / 0.173);
      --blue-a6: color(display-p3 0.0039 0.2471 0.9373 / 0.244);
      --blue-a7: color(display-p3 0.0039 0.2314 0.8706 / 0.33);
      --blue-a8: color(display-p3 0.0039 0.2039 0.8078 / 0.448);
      --blue-a9: color(display-p3 0 0.1529 0.7725 / 0.726);
      --blue-a10: color(display-p3 0 0.1294 0.6549 / 0.761);
      --blue-a11: color(display-p3 0 0.1333 0.6667 / 0.746);
      --blue-a12: color(display-p3 0 0.0588 0.251 / 0.871);

      --blue-contrast: #fff;
      --blue-surface: color(display-p3 0.9608 0.9725 1 / 0.8);
      --blue-indicator: oklch(54.3% 0.1913 266.8);
      --blue-track: oklch(54.3% 0.1913 266.8);
    }
  }
}"
`;

exports[`Basic functionality should generate background value 1`] = `
":root, .light, .light-theme, .radix-themes {
  --color-background: #111;
}"
`;

exports[`Basic functionality should generate grey scales 1`] = `
":root, .light, .light-theme {
  --gray-1: #030304;
  --gray-2: #000;
  --gray-3: #000;
  --gray-4: #000;
  --gray-5: #000;
  --gray-6: #000;
  --gray-7: #000;
  --gray-8: #000;
  --gray-9: #000;
  --gray-10: #000;
  --gray-11: #000;
  --gray-12: #1e1f24;

  --gray-a1: #000001d2;
  --gray-a2: #000000;
  --gray-a3: #000000;
  --gray-a4: #000000;
  --gray-a5: #000000;
  --gray-a6: #000000;
  --gray-a7: #000000;
  --gray-a8: #000000;
  --gray-a9: #000000;
  --gray-a10: #000000;
  --gray-a11: #000000;
  --gray-a12: #afbbf815;

  --gray-contrast: #fff;
  --gray-surface: #ffffffcc;
  --gray-indicator: #000;
  --gray-track: #000;
}

@supports (color: color(display-p3 1 1 1)) {
  @media (color-gamut: p3) {
    :root, .light, .light-theme {
      --gray-1: oklch(10% 0.0015 277.7);
      --gray-2: oklch(2.3% 0.003 277.7);
      --gray-3: oklch(-7.1% 0.0045 277.7);
      --gray-4: oklch(-16.2% 0.0061 277.7);
      --gray-5: oklch(-24.8% 0.0077 277.7);
      --gray-6: oklch(-33.1% 0.0093 277.7);
      --gray-7: oklch(-42.2% 0.0117 277.7);
      --gray-8: oklch(-52.9% 0.016 277.7);
      --gray-9: oklch(-71% 0.0165 277.7);
      --gray-10: oklch(-74.8% 0.0161 277.7);
      --gray-11: oklch(-78% 0.0139 277.7);
      --gray-12: oklch(24.1% 0.0099 277.7);

      --gray-a1: color(display-p3 0 0 0.0039 / 0.824);
      --gray-a2: color(display-p3 0 0 0);
      --gray-a3: color(display-p3 0 0 0);
      --gray-a4: color(display-p3 0 0 0);
      --gray-a5: color(display-p3 0 0 0);
      --gray-a6: color(display-p3 0 0 0);
      --gray-a7: color(display-p3 0 0 0);
      --gray-a8: color(display-p3 0 0 0);
      --gray-a9: color(display-p3 0 0 0);
      --gray-a10: color(display-p3 0 0 0);
      --gray-a11: color(display-p3 0 0 0);
      --gray-a12: color(display-p3 0.7059 0.7529 1 / 0.08);

      --gray-contrast: #fff;
      --gray-surface: color(display-p3 1 1 1 / 80%);
      --gray-indicator: oklch(-71% 0.0165 277.7);
      --gray-track: oklch(-71% 0.0165 277.7);
    }
  }
}"
`;
